# `r description` 

```{r echo=FALSE}
if(file.exists(path)) {
  test.data <- read.csv(path,
                        header=TRUE,
                        sep="\t",
                        check.names = FALSE,
                        na.strings=c("NA")
                        )
  
  ps <- test.data$p_lrt
  ps.l <- length(ps)
} else {
  ps.l <- 0
}

stopifnot(ps.l > 0)

test.gwas <- data.frame(
  SNP = test.data$rs,
  CHR = as.integer(test.data$chr) + 1,
  BP = as.integer(test.data$ps),
  P = ps
  )

test.gwas <- test.gwas[!is.na(test.gwas$P),]
p.vals.s <- sort(ps)
p.trs.pos <- min(length(p.vals.s), p.maxsel)
p.trs.val <- min(p.vals.s[p.trs.pos], p.threshold)
```

__Manhattan Plot__

```{r, warning=FALSE}
manhattan(test.gwas, cex = .5)
```

__Q-Q Plot__

```{r, warning=FALSE}
qq(ps, main = "Q-Q plot of GWAS p-values")
```

__Significant mutations__
```{r, warning=FALSE}
result <- test.data[,c("ps", "allele1", "allele0", "af", "p_lrt")]
for(m in p.adj.m) {
  result[m] <- p.adjust(ps, m)
}
test.data.filtered <- result[ps <= p.trs.val,]
kable(test.data.filtered, digits = 20)
```